website/blog/2022-03-04-aluminium.md

```diff
@@ -0,0 +1,411 @@
+
+<table>
+<tbody>
+  <tr>
+    <td>Commits since last release</td>
+    <td align="center">197</td>
+  </tr>
+  <tr>
+    <td>Merged PRs</td>
+    <td align="center">94</td>
+  </tr>
+    <tr>
+    <td>Contributors</td>
+    <td align="center">11</td>
+  </tr>
+  <tr>
+    <td>Closed issues</td>
+    <td align="center"></td>
+  </tr>
+  <tr>
+    <td>New features</td>
+    <td align="center"></td>
+  </tr>
+</tbody>
+
+</table>
+
+For full details: https://github.com/scalameta/metals/milestone/?closed=1
+
+Metals is a language server for Scala that works with VS Code, Vim, Emacs,
+Sublime Text and Eclipse. Metals is developed at the
+[Scala Center](https://scala.epfl.ch/) and [VirtusLab](https://virtuslab.com)
+with the help from [Lunatech](https://lunatech.com) along with contributors from
+the community.
+
+## TL;DR
+
+Check out [https://scalameta.org/metals/](https://scalameta.org/metals/), and
+give Metals a try!
+
+- Reduce indexing time in monorepos on MacOS
+- Display build target info
+- View source jar files as virtual docs
+- Improve completions for Scala 3
+- [Test Explorer] Detect and run single tests for JUnit4
+
+## Reduce indexing time in monorepos on MacOS
+
+Thanks to changes made by [Duhemm](https://github.com/Duhemm), Metals on MacOS can now index monorepo workspace much more faster than before. After changes, we observed even **over 10 times faster** indexing time in the workspaces where only a few submodules are needed to be imported.  
+
+Here are a few examples of indexing improvement.
+
+Before | After
+---|---
+\>10minutes | ~30seconds
+~40seconds  | ~4s
+
+However, take into mind that indexing speedup heavily depends on your repository structure. The more imported modules which don't need to be imported, the greater speed improvement is.
+
+If you are interested in details you can check related [pull request](https://github.com/scalameta/metals/pull/3665).
+
+## Display build target info
+
+Thanks to the [Arthurm1](https://github.com/Arthurm1) Metals are able to display all important information about modules.
+This brand new feature gathers in one view information such as: 
+- javac and scalac options
+- dependent modules
+- projects classpath
+- and many more
+
+![](https://i.imgur.com/XGyJEsl.gif)
+
+## View source jar files as virtual docs
+
+Previously, when the client wanted to browse files in source jars source, these files were extracted and saved in the `metals/readonly/dependencies` directory. With the help of [virtual documents](https://code.visualstudio.com/api/extension-guides/virtual-documents) Metals can show you dependencies in readonly files without unnecessary copying.
+Thanks [Arthurm1](https://github.com/Arthurm1) for this feature!
+
+Together with Metals tab, this feature could be used to browse through your dependencies' sources.
+
+![](https://imgur.com/5ni5ZIG.gif)
+
+## Improve completions for Scala 3
+
+todo
```

Comment:
Are you about [this pr](https://github.com/scalameta/metals/pull/3651)?  It was just a refactoring - no actual changes/improvements.

Reply:
You did newline completions, @tgodzik did keyword completions and I expect both of you to prepare some text + gif ;)

Reply:
https://github.com/scalameta/metals/pull/3629 - You
https://github.com/scalameta/metals/pull/3560 - Tomek 
:)

Reply:
ouch. totally forgot.